Core/ModuleInstaller.cs

```diff
@@ -889,29 +889,41 @@ private void Uninstall(string modName)
         /// <summary>
         /// Takes a collection of directories and adds all parent directories within the GameData structure.
         /// </summary>
-        /// <param name="directories">The collection of directory path strings to examine </param>
-        private HashSet<string> AddParentDirectories(HashSet<string> directories )
+        /// <param name="directories">The collection of directory path strings to examine</param>
+        public HashSet<string> AddParentDirectories(HashSet<string> directories)
         {
-            var newDirectories = new HashSet<string>();
-            foreach (string directory in directories)
+            if (directories == null || directories.Count == 0)
             {
-                if ( directory.Contains(ksp.GameData()))
+                return new HashSet<string>();
+            }
+
+            var gameData = KSPPathUtils.NormalizePath(ksp.GameData());
+            return directories
+                .Where(dir => !string.IsNullOrWhiteSpace(dir))
+                // normalize before we deduplicate this list
+                .Select(KSPPathUtils.NormalizePath)
+                // remove any duplicate paths
+                .Distinct()
+                // get rid of paths that are above GameData
+                .Where(dir => dir.Length > gameData.Length)
```

Comment:
This would exclude any directory with a name shorter than 9 characters that was installed to Gameroot?